File: frontend/src/types/company.ts
export interface Company {
  id: string;
  name: string;
  domain: string | null;
  website: string | null;
  careersUrl: string | null;
  sourceId: string;
  sector: string | null;
  stage: string | null;
  enriched: boolean;
  enrichedAt: string | null;
  createdAt: string;
}

File: backend/schemas/company.py
"""Company response schemas."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class CompanyOut(BaseModel):
    id: str
    name: str
    domain: str | None
    website: str | None
    careers_url: str | None
    source_id: str | None
    sector: str | None
    stage: str | None
    headcount: str | None
    description: str | None
    enriched: bool
    enriched_at: datetime | None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)
